modules/payment_example/src/Plugin/Commerce/PaymentGateway/Onsite.php

```diff
@@ -84,6 +84,16 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
   /**
    * {@inheritdoc}
    */
+  public function shouldCapturePaymentByDefault() {
+    // Determine default capture flag from current mode (test / live).
+    // Another way for payment gateways to determine this value could be for
+    // them to evaluate the value of another field specific to this setting.
+    return ($this->configuration['mode'] === 'live');
```

Comment:
I think by default we can just have it return `TRUE` live or sandbox doesn't affect capture immediately. Payment gateways supporting modes can use their own logic here.